appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -805,42 +847,50 @@ Blockly.LexicalVariable.renameParamWithoutRenamingCapturablesInfo = function (so
 }
 
 /**
- * [lyn, 10/27/13]
- * Checks an identifier for validity. Validity rules are a simplified version of Kawa identifier rules.
- * They assume that the YAIL-generated version of the identifier will be preceded by a legal Kawa prefix:
+ * Checks an identifier for validity. Validity rules are a simplified version of
+ * Kawa identifier rules. They assume that the YAIL-generated version of the
+ * identifier will be preceded by a legal Kawa prefix:
  *
  *   <identifier> = <first><rest>*
  *   <first> = letter U charsIn("_$?~@")
  *   <rest> = <first> U digit
  *
  *   Note: an earlier version also allowed characters in "!&%.^/+-*>=<",
  *   but we decided to remove these because (1) they may be used for arithmetic,
- *   logic, and selection infix operators in a future AI text language, and we don't want
- *   things like a+b, !c, d.e to be ambiguous between variables and other expressions.
- *   (2) using chars in "><&" causes HTML problems with getters/setters in flydown menu.
+ *   logic, and selection infix operators in a future AI text language, and we
+ *   don't want things like a+b, !c, d.e to be ambiguous between variables and
+ *   other expressions. (2) using chars in "><&" causes HTML problems with
+ *   getters/setters in flydown menu.
  *
  * First transforms the name by removing leading and trailing whitespace and
  * converting nonempty sequences of internal whitespace to '_'.
- * Returns a result object of the form {transformed: <string>, isLegal: <bool>}, where:
- * result.transformed is the transformed name and result.isLegal is whether the transformed
- * named satisfies the above rules.
+ * @param {string} ident The original identifier. 
+ * @return {!{transformed: string, isLegal: boolean}} A result object where the
+ *     transform property is the passed identifier with all whitespace converted
+ *     to underscores, and the isLegal property describes if the transformed
+ *     identifier contains any illegal characters.
  */
+// TODO: This function is only used by makeLegalIdentifier. Can we combine them?
```

Comment:
I would keep it separate for ease of testing. The compiler will likely inline it for us anyway.

Reply:
Sounds good, removed.